From aa0c385314629a044be1c985e7d94206eaf7cf53 Mon Sep 17 00:00:00 2001
From: David Brown <david.brown@linaro.org>
Date: Tue, 3 May 2016 13:29:46 -0600
Subject: [PATCH 1/2] Allow initrd to be empty

The objtools is unhappy if the initrd is either /dev/null or an empty
file.  Check, in the Makefile, for /dev/null, and if so, create a
small empty file for the initrd.  The kernel will ignore an initrd
that doesn't have the proper header, so this will achieve the same
effect.

Signed-off-by: David Brown <david.brown@linaro.org>
---
 bios/link.mk | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/bios/link.mk b/bios/link.mk
index d8309cd..2182376 100644
--- a/bios/link.mk
+++ b/bios/link.mk
@@ -84,12 +84,19 @@ endif
 
 ifndef BIOS_NSEC_ROOTFS
 $(error BIOS_NSEC_ROOTFS not defined!)
-endif
+else ifeq ($(BIOS_NSEC_ROOTFS),/dev/null)
+$(out-dir)nsec_rootfs.bin: FORCE
+	@echo '  MAKE    $@'
+	@mkdir -p $(dir $@)
+	@rm -f $@
+	$(q) echo 'Empty' > $@
+else
 $(out-dir)nsec_rootfs.bin: $(BIOS_NSEC_ROOTFS) FORCE
 	@echo '  LN      $@'
 	@mkdir -p $(dir $@)
 	@rm -f $@
 	$(q)ln -s $(abspath $<) $@
+endif
 
 $(out-dir)nsec_rootfs.o: $(out-dir)nsec_rootfs.bin FORCE
 	@echo '  OBJCOPY $@'
-- 
2.8.2

